presto-main/src/test/java/com/facebook/presto/type/TestFloatType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+
+import static com.facebook.presto.spi.type.FloatType.FLOAT;
+
+public class TestFloatType
+        extends AbstractTestType
+{
+    public TestFloatType()
+    {
+        super(FLOAT, Float.class, createTestBlock());
+    }
+
+    public static Block createTestBlock()
+    {
+        BlockBuilder blockBuilder = FLOAT.createBlockBuilder(new BlockBuilderStatus(), 30);
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(11.11F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(11.11F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(11.11F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(22.22F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(22.22F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(22.22F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(22.22F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(22.22F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(33.33F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(33.33F));
+        FLOAT.writeLong(blockBuilder, Float.floatToRawIntBits(44.44F));
+        FLOAT.writeFloat(blockBuilder, 11.11F);
```

Comment:
writeFloat should not exist - the only way to write floats is how you have done it above.